reactos/drivers/usb/usbmport.h

```diff
@@ -0,0 +1,716 @@
+  PUSBPORT_WAIT UsbPortWait;
+  PUSBPORT_INVALIDATE_CONTROLLER UsbPortInvalidateController;
+  PUSBPORT_BUG_CHECK UsbPortBugCheck;
+  PUSBPORT_NOTIFY_DOUBLE_BUFFER UsbPortNotifyDoubleBuffer;
+
+  /* Miniport */
+
+  PHCI_REBALANCE_ENDPOINT RebalanceEndpoint;
+  PHCI_FLUSH_INTERRUPTS FlushInterrupts;
+  PHCI_RH_CHIRP_ROOT_PORT RH_ChirpRootPort;
+  PHCI_TAKE_PORT_CONTROL TakePortControl;
+  ULONG Reserved4;
+  ULONG Reserved5;
+} USBPORT_REGISTRATION_PACKET, *PUSBPORT_REGISTRATION_PACKET;
+
+typedef struct _USBPORT_MINIPORT_INTERFACE {
+  PDRIVER_OBJECT DriverObject;
+  LIST_ENTRY DriverLink;
+  PDRIVER_UNLOAD DriverUnload;
+  ULONG Version;
+  USBPORT_REGISTRATION_PACKET Packet;
+} USBPORT_MINIPORT_INTERFACE, *PUSBPORT_MINIPORT_INTERFACE;
+
+C_ASSERT(sizeof(USBPORT_MINIPORT_INTERFACE) == 336);
+
+typedef struct _USBPORT_ENDPOINT_PROPERTIES {
+  USHORT DeviceAddress;
+  USHORT EndpointAddress;
+  USHORT TotalMaxPacketSize; // TransactionPerMicroframe * MaxPacketSize
+  UCHAR Period;
+  UCHAR Reserved1;
+  USB_DEVICE_SPEED DeviceSpeed;
+  ULONG UsbBandwidth;
+  ULONG ScheduleOffset;
+  ULONG TransferType;
+  ULONG Direction;
+  ULONG BufferVA;
+  ULONG BufferPA;
+  ULONG BufferLength;
+  ULONG Reserved3;
+  ULONG MaxTransferSize;
+  USHORT HubAddr;
+  USHORT PortNumber;
+  UCHAR InterruptScheduleMask;
+  UCHAR SplitCompletionMask;
+  UCHAR TransactionPerMicroframe; // 1 + additional transactions. Total: from 1 to 3)
+  UCHAR Reserved4;
+  ULONG MaxPacketSize;
+  ULONG Reserved6;
+} USBPORT_ENDPOINT_PROPERTIES, *PUSBPORT_ENDPOINT_PROPERTIES;
+
+C_ASSERT(sizeof(USBPORT_ENDPOINT_PROPERTIES) == 64);
+
+typedef struct _USBPORT_SCATTER_GATHER_ELEMENT {
+  PHYSICAL_ADDRESS SgPhysicalAddress;
+  ULONG Reserved1;
+  ULONG SgTransferLength;
+  ULONG SgOffset;
+  ULONG Reserved2;
+} USBPORT_SCATTER_GATHER_ELEMENT, *PUSBPORT_SCATTER_GATHER_ELEMENT;
+
+C_ASSERT(sizeof(USBPORT_SCATTER_GATHER_ELEMENT) == 24);
+
+typedef struct _USBPORT_SCATTER_GATHER_LIST {
+  ULONG Flags;
+  ULONG_PTR CurrentVa;
+  PVOID MappedSystemVa;
+  ULONG SgElementCount;
+  USBPORT_SCATTER_GATHER_ELEMENT SgElement[1];
+} USBPORT_SCATTER_GATHER_LIST, *PUSBPORT_SCATTER_GATHER_LIST;
+
+C_ASSERT(sizeof(USBPORT_SCATTER_GATHER_LIST) == 40);
+
+typedef struct _USBPORT_TRANSFER_PARAMETERS {
+  ULONG TransferFlags;
+  ULONG TransferBufferLength;
+  ULONG TransferCounter;
+  ULONG Reserved1;
+  ULONG Reserved2;
+  USB_DEFAULT_PIPE_SETUP_PACKET SetupPacket;
+} USBPORT_TRANSFER_PARAMETERS, *PUSBPORT_TRANSFER_PARAMETERS;
+
+C_ASSERT(sizeof(USBPORT_TRANSFER_PARAMETERS) == 28);
+
+typedef struct _USBPORT_ROOT_HUB_DATA {
+  ULONG NumberOfPorts;
+  ULONG HubCharacteristics;
+  ULONG PowerOnToPowerGood;
+  ULONG HubControlCurrent;
+} USBPORT_ROOT_HUB_DATA, *PUSBPORT_ROOT_HUB_DATA;
+
+C_ASSERT(sizeof(USBPORT_ROOT_HUB_DATA) == 16);
+
+/* Hub port status and port status change bits.
+   See USB 2.0 spec Table 11-19 and Table 11-20.
+*/
+
+typedef union _USB_PORT_STATUS {
```

Comment:
You're right, we're missing several things in that file. I'll fix that up in trunk today.